### PR TITLE
Use non-blocking zmq Poller

### LINKED ISSUE #1022: zmq.Poller is not async
The line https://github.com/jupyter/jupyter_client/blob/0a19278ed8d4090080a44760498a38ad9d4e0d99/jupyter_client/client.py#L533 makes the code blocking.

We should use `zmq.asyncio.Poller`.

### PR DESCRIPTION
**Pull request by fcollonval**
Fixes #1022

### CODE REVIEW

**Review by davidbrochart — APPROVED** [review]
Failures are unrelated, see #1024.

**Comment by ianthomas23** [comment]
Pinning `pytest<8.2` helps with the test failures.

File: jupyter_client/channels.py [hunk 1]
```diff
@@ -220,7 +220,7 @@ def _recv(self, **kwargs: t.Any) -> t.Dict[str, t.Any]:
         ident, smsg = self.session.feed_identities(msg)
         return self.session.deserialize(smsg)
 
-    def get_msg(self, timeout: t.Optional[float] = None) -> t.Dict[str, t.Any]:
+    def get_msg(self, timeout: t.Optional[int] = None) -> t.Dict[str, t.Any]:
```
Comment (minrk) [inline jupyter_client/channels.py:223]: This should accept a float. `get_msg(timeout=0.25)` is valid. The cast to int should happen after multiplying by 1000 to get ms.

**Review by minrk — COMMENTED** [review]
👍 to the headline Poller-related change, but the unrelated get_msg type change isn't right, I think.

**Review by minrk — APPROVED** [review]

**Comment by krassowski** [comment]
It would be great to see this released :) Is the timeout in one of the Windows tests related?

**Comment by davidbrochart** [comment]
I don't think it's related. I'd like to merge but I cannot bypass the rules. Maybe @blink1073 can help?

**Pull request merged**